udata/sentry.py

```diff
@@ -11,18 +12,26 @@
 log = logging.getLogger(__name__)
 
 RE_DSN = re.compile(
-    r'(?P<scheme>https?)://(?P<client_id>[0-9a-f]+):[0-9a-f]+'
-    '@(?P<domain>.+)/(?P<site_id>\d+)')
+    r'(?P<scheme>https?)://(?P<client_id>[0-9a-f]+)(?::(?P<secret>[0-9a-f]+))?'
+    r'@(?P<domain>.+)/(?P<site_id>\d+)')
+
+SECRET_DSN_DEPRECATED_MSG = 'DSN with secret is deprecated, use a public DSN instead'
 
 # Controlled exceptions that Sentry should ignore
 IGNORED_EXCEPTIONS = HTTPException, PermissionDenied, UploadProgress
 
-
 def public_dsn(dsn):
-    '''Transform a standard Sentry DSN into a public one'''
+    '''Check if DSN is public or raise a warning and turn it into a public one'''
     m = RE_DSN.match(dsn)
     if not m:
         log.error('Unable to parse Sentry DSN')
+
+    if not m["secret"]:
+        return dsn
+
+    log.warning(SECRET_DSN_DEPRECATED_MSG)
+    warnings.warn(SECRET_DSN_DEPRECATED_MSG, category=DeprecationWarning, stacklevel=2)
```

Comment:
what does `stacklevel=2` do?

Reply:
It is an artifact, not relevant here. It unwinds to the previous caller for the stack trace. Here it would show `init_app` instead of `public_dsn` stack trace level. I'm removing it.